File: common-kotlin/src/main/kotlin/org/hexalite/network/protobuf/impl/entity/Role.kt
package org.hexalite.network.protobuf.impl.entity

import kotlinx.serialization.SerialName
import org.hexalite.network.common.math.Color

@kotlinx.serialization.Serializable
data class Role(
    val id: String,
    @SerialName("unicode_characters")
    val unicodeCharacters: String,
    @SerialName("tab_list_index")
    val tabListIndex: Int,
    val color: Color,
    val permissions: List<Permission>,
)

@kotlinx.serialization.Serializable
inline class Permission(val name: String)

File: rest-webserver/src/entity/role.rs
pub use crate::definitions::entity::Role;
use crate::definitions::returned::ReturnedRole;

impl From<&Role> for ReturnedRole {
    fn from(role: &Role) -> Self {
        Self {
            id: role.id.clone(),
            unicode_character: role.unicode_character.clone(),
            color: role.color.clone(),
            tab_list_index: role.tab_list_index.clone(),
        }
    }
}
